packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/ReactTextPropertiesResolver.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.sessionreplay
+
+import android.view.Gravity
+import android.widget.TextView
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.reactnative.sessionreplay.extensions.convertToDensityNormalized
+import com.datadog.reactnative.sessionreplay.utils.DrawableUtils
+import com.datadog.reactnative.sessionreplay.utils.ReactViewBackgroundDrawableUtils
+import com.datadog.reactnative.sessionreplay.utils.ReflectionUtils
+import com.datadog.reactnative.sessionreplay.utils.formatAsRgba
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.views.text.TextAttributes
+import com.facebook.react.views.view.ReactViewBackgroundDrawable
+import java.util.Locale
+
+internal class ReactTextPropertiesResolver(
+    private val reactContext: ReactContext,
+    private val uiManagerModule: UIManagerModule,
+    private val reflectionUtils: ReflectionUtils = ReflectionUtils(),
+    private val reactViewBackgroundDrawableUtils: ReactViewBackgroundDrawableUtils =
+        ReactViewBackgroundDrawableUtils(),
+    private val drawableUtils: DrawableUtils = DrawableUtils()
+): TextPropertiesResolver {
+    override fun addReactNativeProperties(
+        originalWireframe: MobileSegment.Wireframe.TextWireframe,
+        view: TextView,
+        pixelDensity: Float,
+    ): MobileSegment.Wireframe.TextWireframe {
+        val textWireframe = resolveDrawableProperties(view, pixelDensity, originalWireframe)
+
+        val shadowNodeWrapper: ShadowNodeWrapper =
+            ShadowNodeWrapper.getShadowNodeWrapper(
+                reactContext = reactContext,
+                uiManagerModule = uiManagerModule,
+                reflectionUtils = reflectionUtils,
+                viewId = view.id) ?: return textWireframe
+
+        val textStyle = resolveTextStyle(textWireframe, pixelDensity, shadowNodeWrapper)
+        val textPosition = textWireframe.textPosition
+        val padding = textPosition?.padding
+        val alignment = resolveTextAlignment(view, textWireframe)
+
+        return textWireframe.copy(
```

Comment:
why do you create a copy here ? Couldn't we just resolve all the properties first and then after create the instance ?

Reply:
I'll change to remove the copy in the drawable method and return pairs of values so we only copy the wireframe once. Copying the wireframe at least once is necessary, because the initial wireframe comes from the superclass map so it's either to do this or to duplicate the logic.